article/app/views/support/structuredData/BlogPosting.scala

```diff
@@ -0,0 +1,59 @@
+package views.support.structuredData
+
+import common.LinkTo
+import controllers.LiveBlogPage
+import model.liveblog.BodyBlock
+import org.joda.time.DateTime
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.RequestHeader
+import views.support.Format
+
+object BlogPosting {
+
+  def zulu(date: DateTime)(implicit request: RequestHeader): String = Format(date, "yyyy-MM-dd'T'HH:mm:ssZ")
+
+  def apply(blog: LiveBlogPage, block: BodyBlock)(implicit request: RequestHeader): JsValue = {
+
+    def blockDate(block: BodyBlock) = block.publishedDate match {
+      case Some(date) => zulu(date)
+      case None => zulu(blog.article.trail.webPublicationDate)
+    }
+
+    def blockBody(block: BodyBlock): String = {
+      if(block.bodyTextSummary.length < 1000){
+        block.bodyTextSummary
+      } else {
+        block.bodyTextSummary.substring(0, 1000)
+      }
+    }
+
+    def blockAuthor(blog: LiveBlogPage, block: BodyBlock): JsValue = {
+
+      val name = block.contributors.headOption match {
+        case Some(id) => blog.article.tags.tags.find(_.id == s"profile/$id").map{ contributorTag =>
+          contributorTag.name
+        }
+        case None => None
+      }
+
+      name match {
+        case Some(thing) => Json.obj("@type" -> "Person", "name" -> thing)
+        case None => Organisation()
+      }
+
+    }
+
+    Json.obj(
+      "@type" -> "BlogPosting",
+      "headline" -> block.title.getOrElse[String](blog.article.trail.headline),
+      "author" -> blockAuthor(blog, block),
+      "publisher" -> Organisation(),
```

Comment:
Unfortunately in order for the validator to pass, the publisher field has to be duplicated in each block (As each update of a live-blog can technically be authored by different authors/organisations).